--- a/Makefile.am
+++ b/Makefile.am
@@ -51,21 +51,27 @@
 
 # Special install rule for the game
 install: install-am
-	make install_gamedata target=@GAME_INSTALLDIR@
+	make install_gamedata target=$(DESTDIR)@GAME_INSTALLDIR@ \
+		statedir=$(DESTDIR)@STATE_INSTALLDIR@
 
 install_gamedata:
 	sh mkinstalldirs $(target)/
 	cp -rv graphics \
                styles \
-	       Atris.Players \
-	       Atris.Scores \
                $(target)/
-	chmod 666 $(target)/Atris.Players
-	chmod 666 $(target)/Atris.Scores
+	-rm -f $(target)/graphics/Makefile* $(target)/styles/Makefile*
+
+	sh mkinstalldirs / $(statedir)/
+	cp -rv Atris.Players \
+	       Atris.Scores \
+               $(statedir)/
+	chmod 666 $(statedir)/Atris.Players $(statedir)/Atris.Scores
+	chown .games $(statedir)/Atris.Players $(statedir)/Atris.Scores
 
 install_gamedocs:
 	sh mkinstalldirs $(target)/
 	cp -rv README* COPYING* Docs*   $(target)/
+	-rm -f $(target)/Docs/Makefile*
 
 # Rule to build tar-gzipped distribution package
 $(PACKAGE)-$(VERSION).tar.gz: dist
@@ -84,7 +90,9 @@
 	mv $(PACKAGE) $(PACKAGE)-$(VERSION)/$(PACKAGE).exe
 	mv $(PACKAGE)-$(VERSION) $(PACKAGE)
 	cp ~/*.dll $(PACKAGE)
-	rm $(PACKAGE)/config* $(PACKAGE)/*.[ch] $(PACKAGE)/Make* $(PACKAGE)/*.m4 $(PACKAGE)/{install-sh,missing,mkinstalldirs,autogen.sh} 
+	rm $(PACKAGE)/config* $(PACKAGE)/*.[ch] $(PACKAGE)/Make* $(PACKAGE)/*.m4
+	rm $(PACKAGE)/install-sh $(PACKAGE)/missing $(PACKAGE)/mkinstalldirs
+	rm $(PACKAGE)/autogen.sh
 	rm $(PACKAGE)/*.in -r $(PACKAGE)/.protos
 	zip -r9 $(PACKAGE)-$(VERSION)-win32.zip $(PACKAGE)
 	rm -r $(PACKAGE)
